### PR TITLE
Support PYTHONDONTWRITEBYTECODE environment variable (GH-93)

### PR DESCRIPTION
**Pull request by joeshannon**
To enable a specific workaround for issue 93.

### CODE REVIEW

File: src/org/python/util/jython.java [hunk 1]
```diff
@@ -723,6 +723,11 @@ private static void addDefaultsFromEnvironment(Properties registry) {
         // Go interactive after script. (PYTHONINSPECT because Python scripts may set it.)
         addDefault(registry, "python.inspect", getenv("PYTHONINSPECT"));
 
+        // PYTHONDONTWRITEBYTECODE
```
Comment (joeshannon) [inline src/org/python/util/jython.java:727]: I put this in addDefaultsFromEnvironment as other env vars are read here but might be better elsewhere as it is not setting a registry entry.

Comment (jeff5) [reply src/org/python/util/jython.java:727]: Let's do it here.

**Comment by jeff5** [comment]
When I last re-worked `jython.java` I inferred that the architectural idea, before my time, had been that environment variables, like command line options, were only to be examined in `org.python.util.jython`, and that everything else should read the registry. This makes sense when you consider that `o.p.u.jython` is just one of the ways an interpreter can be started. I removed some use by `PySystemState` of environment variables, and made it go via the registry instead.

I'm not sure where `Options` was supposed to fit in. `o.p.u.jython` was setting options directly, but also `Options` was reading the registry.

I think a purist would unset `python.writebytecode` in response to `-B` or `PYTHONDONTWRITEBYTECODE` and `sys.dont_write_bytecode` would pick it up from there. However, doing that consistently means a lot of other change, so let's be pragmatic.

**Comment by jeff5** [comment]
Can I bother you to sign a [Contributor Agreement](https://www.python.org/psf/contrib/contrib-form/)? We release under the PSF licence, but because our repos are not in the official Python GitHub organisation (because we were here first!) we cannot yet automate it with the same CLA robot. It asks for a bugs.python.org username, but hat has just been decommissioned and what is needed is your GitHub name @joeshannon .

**Comment by joeshannon** [comment]
Thanks for looking at this. I have just signed the CA.

**Comment by jeff5** [comment]
Thanks Joe. I'll look at 129 too.

**Pull request merged**